client/routes/workflow/components/cytoscape-graph.vue

```diff
@@ -0,0 +1,237 @@
+<template>
+  <div id="cytoscape">
+    <div ref="cy" id="cy"></div>
+  </div>
+</template>
+
+<script>
+import dagre from "cytoscape-dagre";
+import { getEventConnections } from "../helpers/get-event-connections";
+import graphStyles from "../helpers/graph-styles";
+import store from "../../../store/index";
+import cytoscape from "cytoscape";
+import omit from "lodash-es/omit";
+
+cytoscape.use(dagre);
+
+export default {
+  name: "cytoscape-graph",
+  props: ["events"],
+  data() {
+    return {
+      nodes: [],
+      edges: [],
+      styles: graphStyles,
+      parentArray: []
+    };
+  },
+  watch: {
+    selectedEvent(id) {
+      if (id) this.selectNode(id);
+    }
+  },
+  methods: {
+    zoomToNode(node) {
+      let zoom = 1.1,
+        bb = node.boundingBox(),
+        w = cy.width(),
+        h = cy.height(),
+        pan = {
+          x: (w - zoom * (bb.x1 + bb.x2)) / 2,
+          y: (h - zoom * (bb.y1 + bb.y2)) / 2
+        };
+
+      //Pan the graph to view node
+      cy.animate({
+        zoom: 1.1,
+        pan: pan
+      });
+    },
+    updateChildBtn(node) {
+      let nodeData = node.data();
+      if (nodeData.childRoute) {
+        store.commit("childRoute", {
+          route: nodeData.childRoute,
+          btnText: "To child"
+        });
+      } else if (nodeData.newExecutionRunId) {
+        store.commit("childRoute", {
+          route: nodeData.newExecutionRunId,
+          btnText: "Next execution"
+        });
+      } else {
+        store.commit("toggleChildBtn");
+      }
+    },
+    selectNode(id) {
+      //Deselect all previously selected nodes
+      cy.$(":selected").deselect();
+
+      //Mark current node as selected
+      let node = cy.elements("node#" + id);
+      node.select();
+
+      this.updateChildBtn(node);
+      this.zoomToNode(node);
+    },
+    async buildTree() {
+      this.events.forEach(event => {
+        let {
+          parentWorkflowExecution,
+          previousExecutionRunId,
+          newExecutionRunId,
+          status,
+          childRoute
+        } = getEventConnections(event, this.events);
+
+        //We are viewing a child workflow, show parent btn
+        if (previousExecutionRunId) {
+          store.commit("previousExecutionRoute", previousExecutionRunId);
+        } else if (parentWorkflowExecution) {
+          store.commit("parentRoute", parentWorkflowExecution);
+        }
```

Comment:
I think if events data was stored in the store, this would already be known and not need a commit step here. To limit the amount of code change, perhaps prime the store in `history/index.vue` when events are received? This might need a bigger re-architecture that might be out-of-scope for this PR.

Reply:
do you mean in history.vue or in index.vue in in the workflow folder?